### DISCUSSION TITLE
renaming a generic doesn't rename the string

### DISCUSSION THREAD

**Discussion by DetachHead** [microsoft/pylance-release#2720] [Enhancement]
### VS Code version

1.63

### Extension version

2021.11.1422169775

### OS type

Windows

### OS version

10

### Python distribution

python.org

### Python version

3.10

### Language server

Default

### Expected behaviour

the argument passed to the generic is renamed

```py
U = TypeVar("U")
```

### Actual behaviour

only the variable is renamed

```py
U = TypeVar("T")
```

### Steps to reproduce

1. make a generic
    ```python
    from typing import TypeVar
    
    T = TypeVar("T")
    ```
2. put your cursor on the `T` variable, hit f2 and rename it
    ![image](https://user-images.githubusercontent.com/57028336/145321625-e8e1b291-173f-4a44-8a09-95dbe4d01ea3.png)
3. hit enter

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by erictraut** [comment]
The string you provide in the `TypeVar` call is different from the variable you assign it to. By convention they should be the same, and pylance will emit an error if they are not, but they are not part of the same symbol table. The "rename symbol" feature renames all "semantically equivalent" versions of a symbol. It shouldn't therefore touch any string literals.

**Comment by KotlinIsland** [comment]
While that is true, I can't imagine a scenario where someone renames a `TypeVar` and doesn't want to rename the string in the call.

Current behavior is very annoying.

**Comment by DetachHead** [comment]
> By convention they should be the same

according to [the PEP](https://www.python.org/dev/peps/pep-0484/#generics):

> The argument to `TypeVar()` must be a string equal to the variable name to which it is assigned.

this sounds more like a requirement than a convention. mypy also doesn't recognize when the string has a different value:
```py
from typing import TypeVar

T = TypeVar("U")

def foo() -> T: ...
```

```
main.py:3: error: String argument 1 "U" to TypeVar(...) does not match variable name "T"
main.py:5: error: Variable "__main__.T" is not valid as a type
main.py:5: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
Found 2 errors in 1 file (checked 1 source file)
```
https://mypy-play.net/?mypy=latest&python=3.10&gist=988c164a48965c40159ae3c791141f86

so since the result of the rename is completely invalid, i think it's an issue that it doesn't change the string

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
